transforms/helpers/transform-helper.ts

```diff
@@ -193,18 +196,22 @@ function createClassProp(
     decorators = createInstancePropDecorators(j, instanceProp);
   }
 
-  const classProp = withDecorators(
-    withComments(
-      j.classProperty(
-        instanceProp.key,
-        shouldSetValue(instanceProp) ? instanceProp.value : null,
-        null
-      ),
-      instanceProp
-    ),
-    decorators
-  );
-  classProp.computed = instanceProp.computed;
+  if ('existingDecorators' in instanceProp) {
+    decorators = [...(instanceProp.existingDecorators ?? []), ...decorators];
+  }
+
+  const classProp = j.classProperty.from({
+    key: instanceProp.key,
+    value: shouldSetValue(instanceProp) ? instanceProp.value : null,
+    comments: instanceProp.comments ?? null,
+    computed: instanceProp.computed,
+  });
+
+  // @ts-expect-error jscodeshift AST types are incorrect
```

Comment:
Could we just augment them in a `.d.ts` file?

Reply:
I suspect that if they fix the type, the will also fix the builder. So I kinda want to stick with `@ts-expect-error` here so that we get alerted when the type is fixed to update the call to the builder. We _should_ just be able to do:

```ts
const classProp = j.classProperty.from({
    key: instanceProp.key,
    value: shouldSetValue(instanceProp) ? instanceProp.value : null,
    comments: instanceProp.comments ?? null,
    computed: instanceProp.computed,
    decorators // <=== THIS
  });
```

Unfortunately that throws an error currently so we have the hack below.

Reply:
Alternatively we could get rid of almost all of the `jscodeshift` stuff and just use the `@babel` builders and types (jscodeshift's API only cares that you take in raw code and spit out transformed raw code; it doesn't care how you got there but provides parsing and transformation utilities as a convenience), but that seemed like another week+ and 8000-line PR.